lib/ts/recipe/passwordless/components/themes/signInUp/linkSent.tsx

```diff
@@ -67,12 +67,13 @@ const PasswordlessLinkSent: React.FC<LinkSentThemeProps> = (props) => {
                 }, 2000);
             } else {
                 setStatus("ERROR");
-                if (response.status === "GENERAL_ERROR") {
-                    props.onError(response.message);
-                }
             }
         } catch (e) {
             setStatus("ERROR");
+
+            if (STGeneralError.isThisError(e)) {
+                props.onError(e.message);
+            }
```

Comment:
How does this work? We first do `setStatus("ERROR")` which I assume shows something went wrong, and then we show the user specific error message if `STGeneralError.isThisError(e)`?

Reply:
Thats correct, setState("ERROR") indicates that there is an error and in the case of an STGeneralError we want the UI to show the appropriate error message